README.org
* Kodachroma

[[https://badge.fury.io/rb/kodachroma][https://badge.fury.io/rb/kodachroma.svg]]

Kodachroma is a color manipulation and palette generation library. It aims to be
a maintained fork of the excellent [[https://github.com/jfairbank/chroma][Chroma]]. Many thanks to [[https://github.com/jfairbank][Jeremy Fairbank]] for
developing Chroma.

From Chroma's own [[https://github.com/jfairbank/chroma/blob/master/README.md][README]]:

#+begin_quote
It is heavily inspired by and a very close Ruby port of the [[https://bgrins.github.io/TinyColor/][tinycolor.js]]
library. Many thanks to [[http://www.briangrinstead.com/blog/][Brian Grinstead]] for his hard work on that library.
#+end_quote

** Installation

Add this line to your application's Gemfile:

#+BEGIN_SRC ruby
gem 'kodachroma'
#+END_SRC

And then execute:

#+begin_src sh
bundle
#+end_src

Or install it yourself as:

#+begin_src sh
gem install kodachroma
#+end_src

** Differences from Chroma

The string extension that defined =String#paint= has been removed. If you would
like to replicate that behavior using Kodachroma, the following should do the
trick.

#+begin_src ruby
class String
  # Creates {Kodachroma::Color} directly from a string representing a color.
  #
  # @example
  #   'red'.paint
  #   '#f00'.paint
  #   '#ff0000'.paint
  #   'rgb(255, 0, 0)'.paint
  #   'hsl(0, 100%, 50%)'.paint
  #   'hsv(0, 100%, 100%)'.paint
  #
  # @return [Kodachroma::Color]
  def paint
    Kodachroma.paint(self)
  end
end
#+end_src

** Creating Colors

Colors are created via the =Kodachroma.paint= method. It expects any one of
many possible color formats as a string, including names, hexadecimal,
rgb, hsl, and hsv.

#+BEGIN_SRC ruby
Kodachroma.paint 'red'                       # named colors
Kodachroma.paint '#00ff00'                   # 6 character hexadecimal
Kodachroma.paint '#00f'                      # 3 character hexadecimal
Kodachroma.paint 'rgb(255, 255, 0)'          # rgb
Kodachroma.paint 'rgba(255, 255, 0, 0.5)'    # rgba
Kodachroma.paint 'hsl(60, 100%, 50%)'        # hsl with percentages
Kodachroma.paint 'hsl(60, 1, 0.5)'           # hsl with decimals
Kodachroma.paint 'hsla(60, 100%, 50%, 0.5)'  # hsla
Kodachroma.paint 'hsv(60, 100%, 50%)'        # hsv with percentages
Kodachroma.paint 'hsv(60, 1, 0.5)'           # hsv with decimals
Kodachroma.paint 'hsva(60, 100%, 50%, 0.75)' # hsva
#+END_SRC

** Motivation

Kodachroma's major strength is manipulating colors and generating color
palettes, which allows you to easily generate dynamic colors, dynamic
themes for a web application, and more.

** Color manipulation

*** Lighten

Lighten the color by a given amount. Defaults to 10.

#+BEGIN_SRC ruby
Kodachroma.paint('red').lighten     #=> #ff3333
Kodachroma.paint('red').lighten(20) #=> #ff6666
#+END_SRC

*** Brighten

Brighten the color by a given amount. Defaults to 10.

#+BEGIN_SRC ruby
Kodachroma.paint('red').brighten     #=> #ff1a1a
Kodachroma.paint('red').brighten(20) #=> #ff3333
#+END_SRC

*** Darken

Darken the color by a given amount. Defaults to 10.

#+BEGIN_SRC ruby
Kodachroma.paint('red').darken     #=> #cc0000
Kodachroma.paint('red').darken(20) #=> #990000
#+END_SRC

*** Desaturate

Desaturate the color by a given amount. Defaults to 10.

#+BEGIN_SRC ruby
Kodachroma.paint('red').desaturate     #=> #f20d0d
Kodachroma.paint('red').desaturate(20) #=> #e61919
#+END_SRC

*** Saturate

Saturate the color by a given amount. Defaults to 10.

#+BEGIN_SRC ruby
Kodachroma.paint('#123').saturate     #=> #0e2236
Kodachroma.paint('#123').saturate(20) #=> #0a223a
#+END_SRC

*** Grayscale

Convert the color to grayscale.

#+BEGIN_SRC ruby
Kodachroma.paint('green').grayscale #=> #404040

# greyscale is an alias
Kodachroma.paint('red').greyscale   #=> #808080
#+END_SRC

*** Opacity

Set the opacity of the color to a given amount.

#+BEGIN_SRC ruby
Kodachroma.paint('red').opacity(0.3) #=> #ff0000
Kodachroma.paint('red').opacity(0.3).to_rgb #=> 'rgba(255, 0, 0, 0.3)'
#+END_SRC

*** Spin

Spin a given amount in degrees around the hue wheel.

#+BEGIN_SRC ruby
Kodachroma.paint('red').spin(30) #=> #ff8000
Kodachroma.paint('red').spin(60) #=> yellow
Kodachroma.paint('red').spin(90) #=> #80ff00
#+END_SRC

** Generating palettes

Kodachroma's most powerful feature is palette generation. You can use the
default palettes available or even create your own custom palettes.

Palette methods are available via =Color#palette= and by default output
an array of colors. If you want the underlying color strings, you can
pass in the desired format via the =:as= option.

**** Available Formats

- name
- rgb
- hex
- hex6 (alias for hex)
- hex3
- hex8 (includes the alpha value in the highest order byte)
- hsl
- hsv

**** Complement

Generate a complement palette.

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.complement            #=> [red, cyan]
Kodachroma.paint('red').palette.complement(as: :name) #=> ['red', 'cyan']
Kodachroma.paint('red').palette.complement(as: :hex)  #=> ['#ff0000', '#00ffff']
#+END_SRC

**** Triad

Generate a triad palette.

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.triad            #=> [red, lime, blue]
Kodachroma.paint('red').palette.triad(as: :name) #=> ['red', 'lime', 'blue']
Kodachroma.paint('red').palette.triad(as: :hex)  #=> ['#ff0000', '#00ff00', '#0000ff']
#+END_SRC

**** Tetrad

Generate a tetrad palette.

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.tetrad
#=> [red, #80ff00, cyan, #7f00ff]

Kodachroma.paint('red').palette.tetrad(as: :name)
#=> ['red', '#80ff00', 'cyan', '#7f00ff']

Kodachroma.paint('red').palette.tetrad(as: :hex)
#=> ['#ff0000', '#80ff00', '#00ffff', '#7f00ff']
#+END_SRC

**** Split complement

Generate a split complement palette.

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.split_complement
#=> [red, #ccff00, #0066ff]

Kodachroma.paint('red').palette.split_complement(as: :name)
#=> ['red', '#ccff00', '#0066ff']

Kodachroma.paint('red').palette.split_complement(as: :hex)
#=> ['#ff0000', '#ccff00', '#0066ff']
#+END_SRC

**** Analogous

Generate an analogous palette. Pass in a =:size= option to specify the
size of the palette (defaults to 6). Pass in a =:slice_by= option to
specify the angle size to slice into the hue wheel (defaults to 30
degrees).

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.analogous
#=> [red, #ff0066, #ff0033, red, #ff3300, #ff6600]

Kodachroma.paint('red').palette.analogous(as: :hex)
#=> ['#f00', '#f06', '#f03', '#f00', '#f30', '#f60']

Kodachroma.paint('red').palette.analogous(size: 3)
#=> [red, #ff001a, #ff1a00]

Kodachroma.paint('red').palette.analogous(size: 3, slice_by: 60)
#=> [red, #ff000d, #ff0d00]
#+END_SRC

**** Monochromatic

Generate a monochromatic palette. Pass in a =:size= option to specify
the size of the palette (defaults to 6).

#+BEGIN_SRC ruby
Kodachroma.paint('red').palette.monochromatic
#=> [red, #2a0000, #550000, maroon, #aa0000, #d40000]

Kodachroma.paint('red').palette.monochromatic(as: :hex)
#=> ['#ff0000', '#2a0000', '#550000', '#800000', '#aa0000', '#d40000']

Kodachroma.paint('red').palette.monochromatic(size: 3)
#=> [red, #550000, #aa0000]
#+END_SRC

** Defining custom palettes

Kodachroma allows you to define your own custom palettes if the default ones
aren't all you're looking for. You can define a custom palette by
calling =Kodachroma.define_palette=, passing in a palette name and
definition block. The definition block uses the color manipulation
methods (i.e. =lighten=, =spin=, etc.) as its DSL. Every DSL call
defines a new color that will be included in the palette. Your seed
color (i.e. the color from which you call the palette method) will be
included as the first color in your palette too.

#+BEGIN_SRC ruby
red = Kodachroma.paint('red')

red.palette.respond_to? :my_palette #=> false

# Define a palette with 5 colors including the seed color
Kodachroma.define_palette :my_palette do
  spin 60
  spin 180
  spin(60).brighten(20) # chain calls as well
  greyscale
end

red.palette.respond_to? :my_palette #=> true

red.palette.my_palette #=> [#ff0000 #ffff00 #00ffff #ffff33 #808080]
#+END_SRC

** Dynamic custom palettes

You can generate custom palettes on the fly too with
=Kodachroma::Color#custom_palette=.

#+BEGIN_SRC ruby
Kodachroma.paint('red').custom_palette do
  spin 60
  spin 180
end

#=> [red, yellow, cyan]
#+END_SRC

** Serializing colors

Colors offer several methods to output to different string color
[[#available-formats][formats]].

| Method      | Description                                                                                                        |
|-------------+--------------------------------------------------------------------------------------------------------------------|
| =to_hsv=    | output to hsv string, outputs hsva if alpha < 1                                                                    |
| =to_hsl=    | output to hsl string, outputs hsla if alpha < 1                                                                    |
| =to_hex=    | output to hex string, optional argument allows 3-character hex output if possible                                  |
| =to_hex8=   | output to 8-character hex string with alpha value in the highest order byte                                        |
| =to_rgb=    | output to rgb string, outputs rgba if alpha < 1                                                                    |
| =to_name=   | output to color name string if available, otherwise ='<unknown>'= or =to_hex= output based on optional arg value   |
| =to_s=      | output to the appropriate string format based on how the color was created, optional arg forces the format         |

#+BEGIN_SRC ruby
# to_hsv
Kodachroma.paint('red').to_hsv                  #=> 'hsv(0, 100%, 100%)'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_hsv #=> 'hsva(0, 100%, 100%, 0.5)'

# to_hsl
Kodachroma.paint('red').to_hsl                  #=> 'hsl(0, 100%, 50%)'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_hsl #=> 'hsla(0, 100%, 50%, 0.5)'

# to_hex
Kodachroma.paint('red').to_hex                  #=> '#ff0000'
Kodachroma.paint('red').to_hex(true)            #=> '#f00'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_hex #=> '#ff0000'
Kodachroma.paint('red').to_hex                  #=> '#ffff0000'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_hex #=> '#80ff0000'

# to_rgb
Kodachroma.paint('red').to_rgb                  #=> 'rgb(255, 0, 0)'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_rgb #=> 'rgb(255, 0, 0, 0.5)'

# to_name
Kodachroma.paint('red').to_name                  #=> 'red'
Kodachroma.paint('#00f').to_name                 #=> 'blue'
Kodachroma.paint('rgba(255, 0, 0, 0.5)').to_name #=> '<unknown>'
Kodachroma.paint('#123').to_name(true)           #=> '#112233'

# to_s
Kodachroma.paint('red').to_s             #=> 'red'
Kodachroma.paint('rgb(255, 0, 0)').to_s  #=> 'rgb(255, 0, 0)'
Kodachroma.paint('#f00').to_s            #=> '#f00'
Kodachroma.paint('#80ff0000').to_s(:rgb) #=> 'rgba(255, 0, 0, 0.5)'
#+END_SRC

** Other methods

Colors also have a few other helper methods:

| Method         | Description                                              |
|----------------+----------------------------------------------------------|
| =dark?=        | is the color dark?                                       |
| =light?=       | is the color light?                                      |
| =alpha=        | retrieve the alpha value                                 |
| =brightness=   | calculate the brightness as a number between 0 and 255   |
| =complement=   | return the complementary color                           |

#+BEGIN_SRC ruby
# dark?
Kodachroma.paint('red').dark?    #=> true
Kodachroma.paint('yellow').dark? #=> false

# light?
Kodachroma.paint('red').light?    #=> false
Kodachroma.paint('yellow').light? #=> true

# alpha
Kodachroma.paint('red').alpha                #=> 1.0
Kodachroma.paint('rgba(0, 0, 0, 0.5)').alpha #=> 0.5

# brightness
Kodachroma.paint('red').brightness    #=> 76.245
Kodachroma.paint('yellow').brightness #=> 225.93
Kodachroma.paint('white').brightness  #=> 255.0
Kodachroma.paint('black').brightness  #=> 0.0

# complement
Kodachroma.paint('red').complement #=> cyan
#+END_SRC

** Trivia

The name "Kodachroma" is inspired by Paul Simon's lovely song "Kodachrome" and
my desire to keep "chroma" in the name of the gem so people searching for the
original gem might find this fork.


CHANGELOG.org
* Kodachroma changelog

** 1.0.2

Fix issue with HSL/HSV values with percent component. Contributed by
[[https://github.com/DEfusion][DEfusion]] in
[[https://github.com/evanleck/kodachroma/pull/3][kodachroma#3]] fixing
[[https://github.com/evanleck/kodachroma/issues/2][kodachroma#2]].

** 1.0.1

Fix eight digit hexidecimal conversion to RGB. Fixes
[[https://github.com/jfairbank/chroma/pull/30][chroma#30]].

** 1.0.0

Initial release of Kodachroma which is effectively a copy and rename of the
excellent [[https://github.com/jfairbank/chroma][Chroma]].

The only breaking change is that the string extension that defined
=String#paint= has been removed. If you would like to replicate that behavior
using Kodachroma, the following should do the trick.

#+begin_src ruby
class String
  # Creates {Kodachroma::Color} directly from a string representing a color.
  #
  # @example
  #   'red'.paint
  #   '#f00'.paint
  #   '#ff0000'.paint
  #   'rgb(255, 0, 0)'.paint
  #   'hsl(0, 100%, 50%)'.paint
  #   'hsv(0, 100%, 100%)'.paint
  #
  # @return [Kodachroma::Color]
  def paint
    Kodachroma.paint(self)
  end
end
#+end_src

* Chroma changelog

/These are the original changes from Chroma./

** 0.2.0

*** NEW =.opacity= Method

Added an =.opacity= method to set the opacity of a color. (credit
[@matildasmeds](https://github.com/matildasmeds))

#+BEGIN_SRC ruby
'red'.paint.opacity(0.3).to_rgb #=> 'rgba(255, 0, 0, 0.3)'
#+END_SRC

--------------

** 0.1.0 - 2016-05-26

*** Dynamic Custom Palettes

You can generate custom palettes on the fly without predefining them
now. The old way of defining them with a name via
=Chroma.define_palette= still works too.

#+BEGIN_SRC ruby
# New dynamic way
'red'.paint.custom_palette do
  spin 60
  spin 180
end

#=> [red, yellow, cyan]
#+END_SRC

** 0.0.1 - 2015-01-14

*Method Changes:*

- Renamed options for analogous palette method.

  - =:results= -> =:size=
  - =:slices= -> =:slice_by=

- Renamed option for monochromatic palette method.

  - =:results= -> =:size=

*Miscellaneous Changes:*

- Add remaining specs for public API.
- Add "transparent" as color name for =Chroma.paint=.
- Minor API doc example fixes.
- Add public API usage examples to README.

** [v0.0.1.alpha.3] - 2015-01-13

*Bug Fixes:*

- Fix bug where =Color#complement= didn't return a color of the same
  format.
- Fix bug where palettes did not have the same format as the seed color.
  (#16)
- Fix bug where the helper method =bound01= was not always producing
  accurate results for percentages due to integer division.
- Fix bug where a =Color= created from an rgba string
  (e.g. ='rgba(255, 0, 0, 0.5).paint'=) was not serializing to an rgba
  string from =to_s=. (#17)

*Method Changes:*

- Add =Color#format= method to return =@format= instance variable.
- Change arguments for =analogous= and =monochromatic= to option
  arguments.
- Add ability to output palette as an array of color format strings via
  the =:as= option. (#10)
- On =Color= rename =greyscale= to =grayscale= and alias =greyscale=
  back to =grayscale=.

*Miscellaneous Changes:*

- Introduced custom errors and replaced =raise= calls with them.
- Added API doc headers. (#4)

** 0.0.1.alpha.2 - 2015-01-13

*Bug Fixes:*

- Fixed bug with number of arguments passed to generator classes in
  RgbGenerator. (#1)
- Make =FromHexStringValues.from_hex8= take alpha as second parameter
  instead of last. Fixes incorrect color generation from hex8. (#6)
- Ensure that string serialization rounds the alpha value where
  applicable. (#7)
- Fix bug where =to_s= and =inspect= would return =<unknown>= instead of
  using hex if the format was =:name= and the named color could not be
  found. (#2)
- Fix bug where =Color= equality wasn't implemented. (#12)
- Fix bug where passing in an instance of =Hsl= or =Hsv= to =Color.new=
  caused their values to get changed. (#11)
- Fix bug with =Color= equality being off due to floating point math.
  (#13)
- Fix bug where =Color= instances generated from hsla and hsva strings
  had the wrong alpha value. (#15)

*Method Changes:*

- Add optional =hex_for_unknown= parameter to
  =Color::Serializers#to_name=. If true, it allows =to_name= to default
  to hex string if name is not found instead of returning ='<unknown>'=.
  (#2)
- Add missing conversion methods to converters
  (a12244f0d81c9480490cfb8a472993f54dd9fbd2)
- Add equality (=eql?= and ====) methods to =Color= class and
  =ColorModes= classes. (#12, #13)
- Add =Chroma.define_palette= for defining custom palettes. (#9)
- Add =Color#paint= method for returning itself. (#14)
- Tweak =Color= serialization method names. Switched to this naming
  primarily to drop the =*_s= on the string serialization methods.

  - =to_hsv= -> =hsv=
  - =to_hsv_s= -> =to_hsv=
  - =to_hsl= -> =hsl=
  - =to_hsl_s= -> =to_hsl=
  - =to_hex= -> =to_basic_hex= (made private)
  - =to_hex_s= -> =to_hex=
  - =to_hex8= -> =to_basic_hex8= (made private)
  - =to_hex8_s= -> =to_hex8=
  - =to_rgb= -> =rgb= (moved attr_reader to serializers and made public)
  - =to_rgb_s= -> =to_rgb=
  - Removed =to_name_s= alias

** 0.0.1.alpha.1 - 2015-01-11

- Initial release


lib/kodachroma/named_colors.rb
# frozen_string_literal: true
module Kodachroma
  NAMED_COLORS = {
    'aliceblue' => 'f0f8ff',
    'antiquewhite' => 'faebd7',
    'aqua' => '0ff',
    'aquamarine' => '7fffd4',
    'azure' => 'f0ffff',
    'beige' => 'f5f5dc',
    'bisque' => 'ffe4c4',
    'black' => '000',
    'blanchedalmond' => 'ffebcd',
    'blue' => '00f',
    'blueviolet' => '8a2be2',
    'brown' => 'a52a2a',
    'burlywood' => 'deb887',
    'burntsienna' => 'ea7e5d',
    'cadetblue' => '5f9ea0',
    'chartreuse' => '7fff00',
    'chocolate' => 'd2691e',
    'coral' => 'ff7f50',
    'cornflowerblue' => '6495ed',
    'cornsilk' => 'fff8dc',
    'crimson' => 'dc143c',
    'cyan' => '0ff',
    'darkblue' => '00008b',
    'darkcyan' => '008b8b',
    'darkgoldenrod' => 'b8860b',
    'darkgray' => 'a9a9a9',
    'darkgreen' => '006400',
    'darkgrey' => 'a9a9a9',
    'darkkhaki' => 'bdb76b',
    'darkmagenta' => '8b008b',
    'darkolivegreen' => '556b2f',
    'darkorange' => 'ff8c00',
    'darkorchid' => '9932cc',
    'darkred' => '8b0000',
    'darksalmon' => 'e9967a',
    'darkseagreen' => '8fbc8f',
    'darkslateblue' => '483d8b',
    'darkslategray' => '2f4f4f',
    'darkslategrey' => '2f4f4f',
    'darkturquoise' => '00ced1',
    'darkviolet' => '9400d3',
    'deeppink' => 'ff1493',
    'deepskyblue' => '00bfff',
    'dimgray' => '696969',
    'dimgrey' => '696969',
    'dodgerblue' => '1e90ff',
    'firebrick' => 'b22222',
    'floralwhite' => 'fffaf0',
    'forestgreen' => '228b22',
    'fuchsia' => 'f0f',
    'gainsboro' => 'dcdcdc',
    'ghostwhite' => 'f8f8ff',
    'gold' => 'ffd700',
    'goldenrod' => 'daa520',
    'gray' => '808080',
    'green' => '008000',
    'greenyellow' => 'adff2f',
    'grey' => '808080',
    'honeydew' => 'f0fff0',
    'hotpink' => 'ff69b4',
    'indianred' => 'cd5c5c',
    'indigo' => '4b0082',
    'ivory' => 'fffff0',
    'khaki' => 'f0e68c',
    'lavender' => 'e6e6fa',
    'lavenderblush' => 'fff0f5',
    'lawngreen' => '7cfc00',
    'lemonchiffon' => 'fffacd',
    'lightblue' => 'add8e6',
    'lightcoral' => 'f08080',
    'lightcyan' => 'e0ffff',
    'lightgoldenrodyellow' => 'fafad2',
    'lightgray' => 'd3d3d3',
    'lightgreen' => '90ee90',
    'lightgrey' => 'd3d3d3',
    'lightpink' => 'ffb6c1',
    'lightsalmon' => 'ffa07a',
    'lightseagreen' => '20b2aa',
    'lightskyblue' => '87cefa',
    'lightslategray' => '789',
    'lightslategrey' => '789',
    'lightsteelblue' => 'b0c4de',
    'lightyellow' => 'ffffe0',
    'lime' => '0f0',
    'limegreen' => '32cd32',
    'linen' => 'faf0e6',
    'magenta' => 'f0f',
    'maroon' => '800000',
    'mediumaquamarine' => '66cdaa',
    'mediumblue' => '0000cd',
    'mediumorchid' => 'ba55d3',
    'mediumpurple' => '9370db',
    'mediumseagreen' => '3cb371',
    'mediumslateblue' => '7b68ee',
    'mediumspringgreen' => '00fa9a',
    'mediumturquoise' => '48d1cc',
    'mediumvioletred' => 'c71585',
    'midnightblue' => '191970',
    'mintcream' => 'f5fffa',
    'mistyrose' => 'ffe4e1',
    'moccasin' => 'ffe4b5',
    'navajowhite' => 'ffdead',
    'navy' => '000080',
    'oldlace' => 'fdf5e6',
    'olive' => '808000',
    'olivedrab' => '6b8e23',
    'orange' => 'ffa500',
    'orangered' => 'ff4500',
    'orchid' => 'da70d6',
    'palegoldenrod' => 'eee8aa',
    'palegreen' => '98fb98',
    'paleturquoise' => 'afeeee',
    'palevioletred' => 'db7093',
    'papayawhip' => 'ffefd5',
    'peachpuff' => 'ffdab9',
    'peru' => 'cd853f',
    'pink' => 'ffc0cb',
    'plum' => 'dda0dd',
    'powderblue' => 'b0e0e6',
    'purple' => '800080',
    'rebeccapurple' => '663399',
    'red' => 'f00',
    'rosybrown' => 'bc8f8f',
    'royalblue' => '4169e1',
    'saddlebrown' => '8b4513',
    'salmon' => 'fa8072',
    'sandybrown' => 'f4a460',
    'seagreen' => '2e8b57',
    'seashell' => 'fff5ee',
    'sienna' => 'a0522d',
    'silver' => 'c0c0c0',
    'skyblue' => '87ceeb',
    'slateblue' => '6a5acd',
    'slategray' => '708090',
    'slategrey' => '708090',
    'snow' => 'fffafa',
    'springgreen' => '00ff7f',
    'steelblue' => '4682b4',
    'tan' => 'd2b48c',
    'teal' => '008080',
    'thistle' => 'd8bfd8',
    'tomato' => 'ff6347',
    'turquoise' => '40e0d0',
    'violet' => 'ee82ee',
    'wheat' => 'f5deb3',
    'white' => 'fff',
    'whitesmoke' => 'f5f5f5',
    'yellow' => 'ff0',
    'yellowgreen' => '9acd32'
  }.freeze

  INVERTED_NAMED_COLORS = NAMED_COLORS.invert.freeze
end
